org.eclipse.wb.swing/src/org/eclipse/wb/internal/swing/preferences/colorchooser/ColorChooserPreferences.java

```diff
@@ -0,0 +1,206 @@
+    setPersistedPreferencesToUI();
+    return container;
+  }
+
+  //Initialize the check boxes to the persisted preferences
+  private void setPersistedPreferencesToUI() {
+    cbxIncludeCustomColor.setSelection(
+        prefs.getBoolean(
+            IColorChooserPreferenceConstants.P_CUSTOM_COLORS,
+            defaultPreferences.get(IColorChooserPreferenceConstants.P_CUSTOM_COLORS)));
+    cbxIncludeSystemColor.setSelection(
+        prefs.getBoolean(
+            IColorChooserPreferenceConstants.P_SYSTEM_COLORS,
+            defaultPreferences.get(IColorChooserPreferenceConstants.P_SYSTEM_COLORS)));
+    cbxIncludeAwtColors.setSelection(
+        prefs.getBoolean(
+            IColorChooserPreferenceConstants.P_AWT_COLORS,
+            defaultPreferences.get(IColorChooserPreferenceConstants.P_AWT_COLORS)));
+    cbxIncludeSwingColors.setSelection(
+        prefs.getBoolean(
+            IColorChooserPreferenceConstants.P_SWING_COLORS,
+            defaultPreferences.get(IColorChooserPreferenceConstants.P_SWING_COLORS)));
+    cbxIncludeNamedColors.setSelection(
+        prefs.getBoolean(
+            IColorChooserPreferenceConstants.P_NAMED_COLORS,
+            defaultPreferences.get(IColorChooserPreferenceConstants.P_NAMED_COLORS)));
+    cbxIncludeWebSafeColors.setSelection(
+        prefs.getBoolean(
+            IColorChooserPreferenceConstants.P_WEB_SAFE_COLORS,
+            defaultPreferences.get(IColorChooserPreferenceConstants.P_WEB_SAFE_COLORS)));
+  }
+
+  @Override
+  protected void performApply() {
+    try {
+      prefs.putBoolean(
+          IColorChooserPreferenceConstants.P_CUSTOM_COLORS,
+          cbxIncludeCustomColor.getSelection());
+      prefs.putBoolean(
+          IColorChooserPreferenceConstants.P_SYSTEM_COLORS,
+          cbxIncludeSystemColor.getSelection());
+      prefs.putBoolean(
+          IColorChooserPreferenceConstants.P_AWT_COLORS,
+          cbxIncludeAwtColors.getSelection());
+      prefs.putBoolean(
+          IColorChooserPreferenceConstants.P_SWING_COLORS,
+          cbxIncludeSwingColors.getSelection());
+      prefs.putBoolean(
+          IColorChooserPreferenceConstants.P_NAMED_COLORS,
+          cbxIncludeNamedColors.getSelection());
+      prefs.putBoolean(
+          IColorChooserPreferenceConstants.P_WEB_SAFE_COLORS,
+          cbxIncludeWebSafeColors.getSelection());
+      prefs.flush();
+      //This ColorPropertyEditor could be null therefore a check is performed.
+      //This method should be in the ColorPropertyEditor itself, but for some reason it seems that
+      //org.osgi.service.prefs.Preferences no longer has a propertyChange method.
+      if (ColorPropertyEditor.INSTANCE != null) {
+        ColorPropertyEditor.reloadColorDialog();
+      }
+    } catch (BackingStoreException e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * This method is called on the "Apply and Close" button
+   */

```

Comment:
If your method does something else than it describes in the superclass then add JavaDoc preferably with the @inheritdoc tag. Otherwise, please remove the javadoc from inherited methods.